lightning/src/chain/keysinterface.rs

```diff
@@ -682,7 +683,8 @@ impl BaseSign for InMemorySigner {
 				} else { return Err(()) }
 			} else { return Err(()) };
 			let mut sighash_parts = bip143::SigHashCache::new(htlc_tx);
-			let sighash = hash_to_message!(&sighash_parts.signature_hash(input, &witness_script, amount, SigHashType::All)[..]);
+			let sighashtype = if self.opt_anchors() { SigHashType::SinglePlusAnyoneCanPay } else { SigHashType::All };
+			let sighash = hash_to_message!(&sighash_parts.signature_hash(input, &witness_script, amount, sighashtype)[..]);
```

Comment:
Are we guaranteed that the first output is the "main" one and the anchors are after that?

Reply:
I don't think I understand your question.  We're signing the resolution tx, which doesn't have any anchors itself ...

When we sign a counterparty `htlc_tx` isn't it always in simple single input/output format? The `SinglePlusAnyoneCanPay` allows the counterparty to use the signature w/ a recomposed "batched" tx with other such inputs?

Reply:
Oh, I was confused on context here, too, why are we changing this? This is creating our own local transaction that is claiming funds available to us. There isn't a second signature so IIUC this would allow a miner/any node to add inputs or outputs to the transaction.

Reply:
I think this code is used to set the witness for the input to the claim tx: https://github.com/lightningdevkit/rust-lightning/blob/main/lightning/src/chain/package.rs#L371 which makes the code correct, iiuc?

Reply:
I think so too, but now think the location referred to in the previous review comment (line 616) is likely wrong because it is the localhtlcsig.  I am noticing that the unit tests aren't specific enough (either way passes at line 616) so I'm going to fix the unit tests and review all of the sighash assignments ...

Reply:
> I think this code is used to set the witness for the input to the claim tx: https://github.com/lightningdevkit/rust-lightning/blob/main/lightning/src/chain/package.rs#L371 which makes the code correct, iiuc?

That seems confusing to me - `package.rs` is always providing our own local signature on a transaction we're about to broadcast. Its never a signature that is validated by our counterparty, because its purely in the on-chain world operating on a closed channel. In that context, we *definitely* need it to be SIGHASH_ALL - the transaction needs to be non-malleable by any relayer or miner.

Reply:
> I think so too, but now think the location referred to in the previous review comment (line 616)

Intuitively, I think 616 is right - we're creating a signature that we send to our counterparty, and our counterparty then goes and stores it for later use in the HTLC transaction they sign + broadcast. In that context, we want our counterparty to be able to add inputs and outputs before they sign it.

Reply:
This call (line 686) is in `sign_counterparty_htlc_transaction`.  Are we saying that we are signing the holder's htlc tx here?  Maybe I'm getting confused w/ the naming?

Reply:
Right, the terminology is all off - both this and the above call are all about counterparty-broadcasted transactions, but this call is about an HTLC output spent by us on a counterparty's commitment transaction. The function docs are a bit clearer -
```
	/// Create a signature for a claiming transaction for a HTLC output on a counterparty's commitment
	/// transaction, either offered or received.
```

Reply:
Ok!  Cool, I'll fix this ...

Reply:
fixed